functions-and-operators/json-functions/json-functions-search.md

```diff
@@ -0,0 +1,360 @@
+
+```sql
+SELECT JSON_KEYS('{"name": {"first": "John", "last": "Doe"}, "type": "Person"}');
+```
+
+```
++---------------------------------------------------------------------------+
+| JSON_KEYS('{"name": {"first": "John", "last": "Doe"}, "type": "Person"}') |
++---------------------------------------------------------------------------+
+| ["name", "type"]                                                          |
++---------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+This shows the keys that are in the `$.name` path of the JSON document.
+
+```sql
+SELECT JSON_KEYS('{"name": {"first": "John", "last": "Doe"}, "type": "Person"}', '$.name');
+```
+
+```
++-------------------------------------------------------------------------------------+
+| JSON_KEYS('{"name": {"first": "John", "last": "Doe"}, "type": "Person"}', '$.name') |
++-------------------------------------------------------------------------------------+
+| ["first", "last"]                                                                   |
++-------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+## [JSON_SEARCH()](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#function_json-search)
+
+The `JSON_SEARCH(json_doc, one_or_all, str)` function searches a JSON document for one or all matches of a string.
+
+Examples:
+
+In the following example, you can search for the first result for `cc`, which is in position 2 of the `a` array.
+
+```sql
+SELECT JSON_SEARCH('{"a": ["aa", "bb", "cc"], "b": ["cc", "dd"]}','one','cc');
+```
+
+```
++------------------------------------------------------------------------+
+| JSON_SEARCH('{"a": ["aa", "bb", "cc"], "b": ["cc", "dd"]}','one','cc') |
++------------------------------------------------------------------------+
+| "$.a[2]"                                                               |
++------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+Now you do the same, but set `sub_or_all` to `all` to get not just the first result, but all results.
+
+```json
+SELECT JSON_SEARCH('{"a": ["aa", "bb", "cc"], "b": ["cc", "dd"]}','all','cc');
+```
+
+```
++------------------------------------------------------------------------+
+| JSON_SEARCH('{"a": ["aa", "bb", "cc"], "b": ["cc", "dd"]}','all','cc') |
++------------------------------------------------------------------------+
+| ["$.a[2]", "$.b[0]"]                                                   |
++------------------------------------------------------------------------+
+1 row in set (0.01 sec)
+```
+
+## [MEMBER OF()](https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#operator_member-of)
+
+The `str MEMBER OF (json_array)` functions tests if the passed value is an element of the `json_array`, it returns `1`. Otherwise, it returns `0`. It returns `NULL` if any of the arguments is `NULL`.
```

Comment:
```suggestion
The `str MEMBER OF (json_array)` function tests if the passed value is an element of the `json_array`, it returns `1`. Otherwise, it returns `0`. It returns `NULL` if any of the arguments is `NULL`.
```